src/main/java/com/aws/greengrass/status/FleetStatusService.java

```diff
@@ -156,46 +161,60 @@ public FleetStatusService(Topics topics, MqttClient mqttClient, DeploymentStatus
-        this.deploymentStatusKeeper.registerDeploymentStatusConsumer(LOCAL,
-                this::deploymentStatusChanged, FLEET_STATUS_SERVICE_TOPICS);
-        this.deploymentStatusKeeper.registerDeploymentStatusConsumer(SHADOW,
-                this::deploymentStatusChanged, FLEET_STATUS_SERVICE_TOPICS);
-        schedulePeriodicFleetStatusDataUpdate(false);
-
         //populating services when kernel starts up
         Instant now = Instant.now();
         this.kernel.orderedDependencies().forEach(greengrassService -> {
             serviceFssTracksMap.put(greengrassService, now);
         });
     }
 
+    @Override
+    public void postInject() {
+        super.postInject();
+
+        deviceConfiguration.onAnyChange((what, node) -> {
+            if (node != null && what.equals(WhatHappened.childChanged)
+                    && deviceConfiguration.provisionInfoNodeChanged(node, this.isFSSSetupComplete.get())) {
+                try {
+                    // Not using isDeviceConfiguredToTalkToCloud() in order to provide the detailed error
+                    // message to user
+                    deviceConfiguration.validate();
+
+                    Topics configurationTopics = deviceConfiguration.getStatusConfigurationTopics();
+                        configurationTopics.lookup(FLEET_STATUS_PERIODIC_PUBLISH_INTERVAL_SEC)
+                            .dflt(DEFAULT_PERIODIC_PUBLISH_INTERVAL_SEC)
+                                .subscribe((why, newv) -> {
+                        int newPeriodicUpdateIntervalSec = Coerce.toInt(newv);
+                        // Do not update the scheduled interval if it is less than the default.
+                        if (newPeriodicUpdateIntervalSec < DEFAULT_PERIODIC_PUBLISH_INTERVAL_SEC) {
+                            return;
+                        }
+                        this.periodicPublishIntervalSec = TestFeatureParameters.retrieveWithDefault(Double.class,
+                                FLEET_STATUS_TEST_PERIODIC_UPDATE_INTERVAL_SEC, newPeriodicUpdateIntervalSec)
+                                .intValue();
+                        if (periodicUpdateFuture != null) {
+                            schedulePeriodicFleetStatusDataUpdate(false);
+                        }
+                    });
+
+                    config.getContext().addGlobalStateChangeListener(this::handleServiceStateChange);
+
+                    this.deploymentStatusKeeper.registerDeploymentStatusConsumer(IOT_JOBS,
+                        this::deploymentStatusChanged, FLEET_STATUS_SERVICE_TOPICS);
+                    this.deploymentStatusKeeper.registerDeploymentStatusConsumer(LOCAL,
+                        this::deploymentStatusChanged, FLEET_STATUS_SERVICE_TOPICS);
+                    this.deploymentStatusKeeper.registerDeploymentStatusConsumer(SHADOW,
+                        this::deploymentStatusChanged, FLEET_STATUS_SERVICE_TOPICS);
+                    schedulePeriodicFleetStatusDataUpdate(false);
```

Comment:
do these add multiple listeners / consumers if the config changes multiple times?

Reply:
good question. will look into this.

Reply:
Added flag to check if initial setup has already been done, and ensured these are only called once.